### PR TITLE
openstack: Deprovision application credentials

### PR DESCRIPTION
**Pull request by pierreprinetti**
Deletion of the ephemeral application credentials was left so far to a separate cleanup job. With this change, each job is supposed to clean after itself.

### CODE REVIEW

File: ci-operator/step-registry/openstack/deprovision/credentials/openstack-deprovision-credentials-commands.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+set -o pipefail
+
+info() {
+	printf '%s: %s\n' "$(date --utc +%Y-%m-%dT%H:%M:%SZ)" "$*"
+}
+
+CLOUDS_YAML="${SHARED_DIR}/clouds.yaml"
+CLOUDS_YAML_UNRESTRICTED="${SHARED_DIR}/clouds-unrestricted.yaml"
+
+if ! [[ -f "$UNRESTRICTED_CLOUDS_YAML" ]]; then
```
Comment (dulek) [inline ci-operator/step-registry/openstack/deprovision/credentials/openstack-deprovision-credentials-commands.sh:14]: `UNRESTRICTED_CLOUDS_YAML` vs `CLOUDS_YAML_UNRESTRICTED` in line 12?

File: ci-operator/step-registry/openstack/rotate/cloud-credentials/openstack-rotate-cloud-credentials-commands.sh [hunk 2]
```diff
@@ -56,7 +56,4 @@ oc wait --timeout=5m --for=condition=Progressing=false $(oc get clusteroperator
 info 'Revoking the credentials that were used so far...'
 delete_application_credential "$ORIGINAL_CLOUDS_YAML"
 
-info 'Revoking the unrestricted credentials...'
-delete_application_credential "$UNRESTRICTED_CLOUDS_YAML"
```
Comment (dulek) [inline ci-operator/step-registry/openstack/rotate/cloud-credentials/openstack-rotate-cloud-credentials-commands.sh:60]: Why is this removed?

**Review by dulek — CHANGES_REQUESTED** [review]
Some remarks inline.

File: ci-operator/step-registry/openstack/deprovision/credentials/openstack-deprovision-credentials-commands.sh (hunk 1 above)
Comment (pierreprinetti) [reply ci-operator/step-registry/openstack/deprovision/credentials/openstack-deprovision-credentials-commands.sh:14]: woops! good catch, thanks.

File: ci-operator/step-registry/openstack/rotate/cloud-credentials/openstack-rotate-cloud-credentials-commands.sh (hunk 2 above)
Comment (pierreprinetti) [reply ci-operator/step-registry/openstack/rotate/cloud-credentials/openstack-rotate-cloud-credentials-commands.sh:60]: We used to delete the unrestricted appcreds in the provisioning step. With this PR, we need them in the deprovisioning step in order to do the cleanup.

**Comment by pierreprinetti** [comment]
/pj-rehearse pull-ci-openshift-cluster-storage-operator-release-4.10-e2e-openstack-parallel pull-ci-openshift-baremetal-runtimecfg-release-4.14-e2e-openstack

**Comment by pierreprinetti** [comment]
The credentials were not found by the deprovisioning step in the rehearse run.
/hold

**Comment by pierreprinetti** [comment]
/pj-rehearse pull-ci-openshift-cluster-storage-operator-release-4.10-e2e-openstack-parallel pull-ci-openshift-baremetal-runtimecfg-release-4.14-e2e-openstack

**Comment by pierreprinetti** [comment]
/pj-rehearse pull-ci-openshift-cluster-storage-operator-release-4.10-e2e-openstack-parallel pull-ci-openshift-baremetal-runtimecfg-release-4.14-e2e-openstack

**Comment by pierreprinetti** [comment]
/pj-rehearse pull-ci-openshift-cluster-storage-operator-release-4.10-e2e-openstack-parallel pull-ci-openshift-baremetal-runtimecfg-release-4.14-e2e-openstack

**Comment by pierreprinetti** [comment]
/pj-rehearse pull-ci-openshift-cluster-storage-operator-release-4.10-e2e-openstack-parallel pull-ci-openshift-baremetal-runtimecfg-release-4.14-e2e-openstack pull-ci-openshift-installer-release-4.14-e2e-openstack-ovn

**Comment by pierreprinetti** [comment]
/pj-rehearse pull-ci-openshift-cluster-storage-operator-release-4.10-e2e-openstack-parallel pull-ci-openshift-baremetal-runtimecfg-release-4.14-e2e-openstack pull-ci-openshift-installer-release-4.14-e2e-openstack-ovn

**Comment by mandre** [comment]
I also want to see a run on a CI job that is not using application credentials.
/pj-rehearse pull-ci-openshift-baremetal-runtimecfg-release-4.11-e2e-openstack

**Review by mandre — COMMENTED** [review]
/lgtm
/approve
@pierreprinetti does your hold still stand?

**Comment by pierreprinetti** [comment]
It does not. Thank you!
/hold cancel

**Comment by dulek** [comment]
@pierreprinetti: Are we fine with rehearsals here and can ack them?

**Comment by pierreprinetti** [comment]
> Are we fine with rehearsals here and can ack them?

:+1:
/pj-rehearse ack
